Lib/dataclasses.py

```diff
@@ -337,6 +341,24 @@ def _tuple_str(obj_name, fields):
     return f'({",".join([f"{obj_name}.{f.name}" for f in fields])},)'
 
 
+def _recursive_repr(user_function):
```

Comment:
Please add a comment that this is using the logic from `reprlib`, and indicate we're duplicating the logic here to avoid importing `reprlib` when it might not be needed elsewhere.